app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -185,30 +187,18 @@ class ComponentsValidator(
   private def messagePrefix(fieldValue: FormComponent) =
     localisation(fieldValue.shortName.getOrElse(fieldValue.label))
 
-  private def validateDateImpl(fieldValue: FormComponent, date: Date)(data: FormDataRecalculated): ValidatedType = {
-    val govDateFormat = DateTimeFormatter.ofPattern("dd MMMM yyyy")
-    val dateWithOffset = (localDate: LocalDate, offset: OffsetDate) =>
-      localDate.plusDays(offset.value.toLong).format(govDateFormat)
+  private def validateDateImpl(fieldValue: FormComponent, date: Date)(data: FormDataRecalculated): ValidatedType =
```

Comment:
Throughout this method, how about extracting some methods to put a name to what's going on & to make it easier to test the pieces?